sui-execution/latest/sui-adapter/src/programmable_transactions/linkage_view.rs

```diff
@@ -279,6 +276,43 @@ impl<'state> LinkageResolver for LinkageView<'state> {
             package.version(),
         )
     }
+
+    pub(crate) fn unpublish_package(&mut self, modules: &[CompiledModule]) {
+        let module_ids: Vec<ModuleId> = modules.iter().map(|m| m.self_id()).collect();
+        for module_id in module_ids {
```

Comment:
Why do you need to collect here?
Could this just be 
```suggestion
        for module_id in modules.iter().map(|m| m.self_id()) {
```

Reply:
I guess so.... lol, we'll change